src/impl/tpls/KokkosBlas3_gemm_tpl_spec_decl.hpp

```diff
@@ -817,4 +817,381 @@ KOKKOSBLAS3_CGEMM_CUBLAS(Kokkos::LayoutRight, Kokkos::LayoutRight,
 }  // namespace KokkosBlas
 #endif  // KOKKOSKERNELS_ENABLE_TPL_CUBLAS
 
+// rocBLAS
+#ifdef KOKKOSKERNELS_ENABLE_TPL_ROCBLAS
+#include <KokkosBlas_tpl_spec.hpp>
+#include <KokkosBlas3_gemm_dotbased_impl.hpp>
+
+namespace KokkosBlas {
+namespace Impl {
+
+#define KOKKOSBLAS3_GEMM_ROCBLAS_DETERMINE_ARGS(LAYOUTA, LAYOUTB, LAYOUTC) \
+  const bool A_t = (transA[0] != 'N') && (transA[0] != 'n');               \
+  const int M    = static_cast<int>(C.extent(0));                          \
+  const int N    = static_cast<int>(C.extent(1));                          \
+  const int K    = static_cast<int>(A.extent(A_t ? 0 : 1));                \
+                                                                           \
+  bool A_is_lr = std::is_same<Kokkos::LayoutRight, LAYOUTA>::value;        \
+  bool B_is_lr = std::is_same<Kokkos::LayoutRight, LAYOUTB>::value;        \
+  bool C_is_lr = std::is_same<Kokkos::LayoutRight, LAYOUTC>::value;        \
+                                                                           \
+  const int AST = A_is_lr ? A.stride(0) : A.stride(1),                     \
+            LDA = AST == 0 ? 1 : AST;                                      \
+  const int BST = B_is_lr ? B.stride(0) : B.stride(1),                     \
+            LDB = BST == 0 ? 1 : BST;                                      \
+  const int CST = C_is_lr ? C.stride(0) : C.stride(1),                     \
+            LDC = CST == 0 ? 1 : CST;                                      \
+                                                                           \
+  rocblas_operation transa, transb;                                        \
```

Comment:
Can we write the small utility function in a separate header so we can reuse it when we do the same work for other kernels.
```c++
static inline rocblas_operation trans_mode_kk_to_rocblas(const char const* kkMode) {
  if(kkMode[0] == 'N' || kk_mode[0] == 'n') {
    return rocblas_operation_none;
  } else if ...
  }
}
```

Reply:
Done